elide-datastore/elide-datastore-hibernate3/src/main/java/com/yahoo/elide/datastores/hibernate3/HibernateTransaction.java

```diff
@@ -248,6 +249,20 @@ public void commit() {
         return new ScrollableIterator(sessionCriteria.scroll(scrollMode));
     }
 
+    @Override
+    public <T> Long getTotalRecords(Class<T> entityClass, FilterScope filterScope) {
+        final Criterion criterion = filterScope.getCriterion(NOT, AND, OR);
+        final CriteriaExplorer criteriaExplorer = new CriteriaExplorer(
+                entityClass, filterScope.getRequestScope(), criterion);
+        final Criteria sessionCriteria = session.createCriteria(entityClass);
+
+        criteriaExplorer.buildCriteria(sessionCriteria, session);
+
+        sessionCriteria.setProjection(Projections.rowCount());
```

Comment:
I think this is equivalent to a `select count(*) ...`, right? So if there was a filter applied, you could select the right set of objects, however, if (by permission) any of those records were filtered out, we would still count them here. I believe the correct value would be something like:

```
this_record_count - total_num_records_filtered
```

This way you're not getting a count for the records which are validly returned by the SQL query, but disallowed by your permissions in Elide.